xs2a-adapter-service-remote/src/main/java/de/adorsys/xs2a/adapter/remote/service/impl/RemoteAccountInformationService.java

```diff
@@ -212,20 +219,29 @@ private ObjectNode createEmptyBody() {
         BookingStatusTO bookingStatus = BookingStatusTO.fromValue(requestParamMap.get(RequestParams.BOOKING_STATUS));
         Boolean deltaList = Boolean.valueOf(requestParamMap.get(RequestParams.DELTA_LIST));
         Boolean withBalance = Boolean.valueOf(requestParamMap.get(RequestParams.WITH_BALANCE));
-        if (requestHeaders.isAcceptJson()) {
-            return (ResponseEntity<T>) client.getTransactionList(accountId, dateFrom, dateTo, entryReferenceFrom, bookingStatus, deltaList, withBalance, requestHeaders.toMap());
 
-        }
         return (ResponseEntity<T>) client.getTransactionListAsString(accountId, dateFrom, dateTo, entryReferenceFrom, bookingStatus, deltaList, withBalance, requestHeaders.toMap());
     }
 
     @Override
     public Response<String> getTransactionListAsString(String accountId, RequestHeaders requestHeaders, RequestParams requestParams) {
         ResponseEntity<String> responseEntity = getTransactionListFromClient(accountId, requestParams, requestHeaders);
 
+        String body;
+
+        if (responseEntity.getBody() instanceof String) {
```

Comment:
Could you explain why you think this instance of check is necessary and when body is going to be a non-string type?

Reply:
you are right, removed